File: kotlin/backjoon/2824.kt
fun main() {
    val br = System.`in`.bufferedReader()
    val INF = 1000000000
    val n = br.readLine().toInt()
    val aList = br.readLine().split(" ").map { it.toLong() }.toMutableList()
    val m = br.readLine().toInt()
    val bList = br.readLine().split(" ").map { it.toLong() }.toMutableList()

    var answer = 1L
    var flag = false
    repeat(n) { i ->
        repeat(m) { j ->
            val g = gcd(aList[i], bList[j])
            answer *= g
            aList[i] /= g
            bList[j] /= g
            if (answer >= INF) {
                answer %= INF
                flag = true
            }
        }
    }
    if (flag) println("$answer".padStart(9, '0')) else println(answer)
}

private fun gcd(a: Long, b: Long): Long {
    var tmp: Long
    var va = a
    var vb = b
    while (vb > 0) {
        tmp = va
        va = vb
        vb = tmp % vb
    }
    return va
}

File: python/backjoon/2824.py
import sys


def gcd(a, b):
    while b > 0:
        tmp = a
        a = b
        b = tmp % b
    return a


n = int(sys.stdin.readline())
a = 1
for i in map(int, sys.stdin.readline().split(" ")):
    a *= i
m = int(sys.stdin.readline())
b = 1
for i in map(int, sys.stdin.readline().split(" ")):
    b *= i
answer = gcd(a, b)

print(str(answer)[-9:])
